### PR TITLE
[AAP-88360] Add AGENTS.md file

### PR DESCRIPTION
**Pull request by john-westcott-iv**
## Description

- Restores the `AGENTS.md` file that was lost during the open-sourcing of aap-gateway to jewel
- Provides AI coding agents (Claude, Cursor, Copilot, etc.) with project context so they can assist developers effectively
- Cleaned up from the upstream aap-gateway version: all internal/proprietary references removed and adapted for the open-source jewel repo

## Type of Change

- [x] Documentation update

## Self-Review Checklist

- [x] I have performed a self-review of my code
- [x] I have added relevant comments to complex code sections
- [x] I have updated documentation where needed
- [x] I have considered the security impact of these changes
- [x] I have considered performance implications
- [x] I have thought about error handling and edge cases
- [x] I have tested the changes in my local environment

### Prerequisites

None - documentation only change.

### Steps to Test

1. Verify `AGENTS.md` contains no internal Red Hat/JIRA references
2. Verify all referenced files (README.md, CONTRIBUTING.md, TESTING.md, docs/) exist in the repo
3. Confirm AI agents can use the file for project context

---
**Note:** This PR was developed with assistance from Claude AI assistant.

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive project guidance covering setup, testing, coding conventions, security, architecture, debugging, and pull request practices.
  * Updated testing documentation with a 90% patch-coverage requirement for new code.
  * Clarified stress-testing instructions and streamlined the required commands.

### CODE REVIEW

**Review by jeffh-oss — APPROVED** [review]

File: AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+- Redis cache
+- PostgreSQL database
+- pytest for testing
+
+**Related Repositories:**
+- This repo may have a checkout of `django-ansible-base` in the `django-ansible-base/` directory - it's a separate git repo with its own tests and configuration
+- `django-ansible-base` is a shared library providing functionality common across open source Ansible services
+
+## Build and Test Commands
+
```
Comment (jay-steurer) [inline AGENTS.md:34]: Can we include ATF somewhere here?

File: AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+### Debugging Methodology
+
+- Always investigate root causes of threading/parallel test issues
+- Stress test fixes with batch runs (`tools/scripts/run_tox_batch.sh`)
+- Prefer systematic debugging over quick hacks
+- Use single-threaded runs to isolate threading problems
+
+## Pull Request Guidelines
+
```
Comment (jay-steurer) [inline AGENTS.md:136]: Remind the agent this is a public repo, all information should be sanitized.

File: AGENTS.md (hunk 1 above)
Comment (john-westcott-iv) [reply AGENTS.md:34]: Should we? Since this is a public repo and ATF is not?

Comment (jay-steurer) [reply AGENTS.md:34]: We can mention functionality testing, end to end testing

**Review by jeffh-oss — APPROVED** [review]
